pkg/apis/elasticsearch/v1/elasticsearch_config.go

```diff
@@ -37,13 +39,14 @@ type ClusterSettings struct {
 
 // Node is the node section in elasticsearch.yml.
 type Node struct {
-	Master     *bool    `config:"master"`
-	Data       *bool    `config:"data"`
-	Ingest     *bool    `config:"ingest"`
-	ML         *bool    `config:"ml"`
-	Transform  *bool    `config:"transform"` // available as of 7.7.0
-	Roles      []string `config:"roles"`     // available as of 7.9.0, takes priority over the other fields if non-nil
-	VotingOnly *bool    `config:"voting_only"`
+	Master              *bool    `config:"master"`
+	Data                *bool    `config:"data"`
+	Ingest              *bool    `config:"ingest"`
+	ML                  *bool    `config:"ml"`
+	Transform           *bool    `config:"transform"`             // available as of 7.7.0
+	RemoteClusterClient *bool    `config:"remote_cluster_client"` // available as of 7.7.0
```

Comment:
We have an inconsistency with our sset pod labelling. We label the `transform` role but we don't label the `voting_only` or the `remote_cluster_role` (which we ignored until this PR). I am not sure what to do about it. Are there node roles that are more important than others? If we consider them all equals we should add labels for all of them on the pods.

Reply:
I think I'm +1 for labelling them all.

Reply:
OK will add another commit here.

Reply:
I added a commit only to realise that there is more amiss #4054